pkg/vip/address.go

```diff
@@ -396,6 +410,88 @@ func (configurator *network) DeleteIP() error {
 		}
 	}
 
+	if configurator.forwardMethod == "masquerade" && configurator.masqueradeMark != "" {
+		if err := configurator.removeIptablesRulesForMasquerade(); err != nil {
+			return errors.Wrap(err, "could not remove iptables masquerade rules ")
+		}
+	}
+
+	return nil
+}
+
+func (configurator *network) addIptablesRulesForMasquerade() error {
+	ver, err := iptables.GetVersion()
+	if err != nil {
+		return errors.Wrap(err, "could not get iptables version")
+	}
+
+	ipt, err := iptables.New(iptables.EnableNFTables(ver.BackendMode == "nft"))
+	if err != nil {
+		return errors.Wrap(err, "could not create iptables client")
+	}
+
+	if err := addMasqueradeRule(ipt, configurator.masqueradeMark, iptablesCommentMarkRule); err != nil {
+		return fmt.Errorf("could not add iptables masquerade rules: %v", err)
+	}
+
+	vip := configurator.address.IP.String()
+	comment := fmt.Sprintf(iptablesComment, vip)
+	if err := addIptablesMarkRuleForVIP(ipt, vip, configurator.masqueradeMark, comment); err != nil {
+		return fmt.Errorf("could not add iptables mark rules: %v", err)
+	}
+
+	return nil
+}
+
+// addIptablesRulesForMasquerade add iptables rules for MASQUERADE
+// insert example
+// sudo iptables -t mangle -I PREROUTING -d 10.1.105.1 -p tcp --dport 6443 -j MARK --set-xmark 0x1119
+// sudo iptables -t nat -I POSTROUTING -m mark --mark 0x1119 -j MASQUERADE
```

Comment:
Hi Lan, just curious:
I use `iptables -t nat -A POSTROUTING -m ipvs --vaddr 192.168.113.200 --vport 8443 -j MASQUERADE` to configure masquerade directly, learning from some materials from internet.
Is there some additional benefit of one of both, if comparing these two methods?

Reply:
I'm not familiar with this rule, can anyone shed some light?